lib/LaTeXML/Post/XSLT.pm

```diff
@@ -76,7 +76,33 @@ sub process {
   if (my $icon = $params{ICON}) {
     # Hmm.... what type? could be various image types
     $params{ICON} = '"' . $self->copyResource($doc, $icon, undef) . '"'; }
+
+  # move to the site directory when applying the stylesheet,
+  # since it can create additional files with <exsl:document>
+  my $sitedir  = $doc->getSiteDirectory;
+  my $orig_cwd = pathname_cwd();
+  pathname_chdir($sitedir);
+
+  # ensure that the stylesheet can only read and write in the site directory
+  # and cannot access the network
+  my $cb = sub {
+    my $dest = pathname_canonical(pathname_absolute($_[1], $sitedir));
+    if (!pathname_is_contained($dest, $sitedir)) {
+      Error('unexpected', 'stylesheet', $doc, "'$dest' is outside of the site directory, denying access");
```

Comment:
```suggestion
      Fatal('unexpected', 'stylesheet', $doc, "'$dest' is outside of the site directory, access denied");
```
since libxslt will crash anyway after this.